### PR TITLE
Api for showing levels

### PR DESCRIPTION
**Pull request by 25shwetak**

### CODE REVIEW

**Comment by 25shwetak** [comment]
[Api for showing levels in boxes](https://trello.com/c/nSiFROnP/39-api-for-showing-levels-in-boxes)

**Comment by ksnehanjali** [comment]
[ESLint and Prettier setup](https://trello.com/c/cLMYKjDK/57-eslint-and-prettier-setup)

File: .rubocop.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ Layout/LineLength:
  Max: 125
 
 Metrics/MethodLength:
- Max: 15
+ Max: 20
```
Comment (tanya-saroha) [inline .rubocop.yml]: Why 20?

File: app/controllers/api/v1/levels_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          attribute = "levels.id as id", "levels.name as name", "levels.description as description", slots[0] + slots[1]
+          level_list = LevelCitySport.select(attribute[0], attribute[1], attribute[2], attribute[3])
+                                     .joins(:level).where(city_sport_id: city_sport.id).uniq
+          if level_list
+            render json: level_list.as_json, status: :ok
+          else
+            render json: {message: "Level doesnot exist"}, status: :not_found
+          end
+        else
+          render json: {message: "City sport doesnot exist"}, status: :not_found
```
Comment (tanya-saroha) [inline app/controllers/api/v1/levels_controller.rb]: doesnot?

File: app/controllers/api/v1/levels_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        if city_sport
+          slots = "#{city_sport.number_of_players} - (SELECT COUNT(*) FROM user_city_sports WHERE ",
+                  "user_city_sports.level_city_sports_id = level_city_sports.id) as free_slots"
+          attribute = "levels.id as id", "levels.name as name", "levels.description as description", slots[0] + slots[1]
+          level_list = LevelCitySport.select(attribute[0], attribute[1], attribute[2], attribute[3])
+                                     .joins(:level).where(city_sport_id: city_sport.id).uniq
+          if level_list
+            render json: level_list.as_json, status: :ok
+          else
+            render json: {message: "Level doesnot exist"}, status: :not_found
```
Comment (tanya-saroha) [inline app/controllers/api/v1/levels_controller.rb]: doesnot?

Comment (tanya-saroha) [inline app/controllers/api/v1/levels_controller.rb]: move to locale

File: app/controllers/api/v1/levels_controller.rb (hunk 2 above)
Comment (tanya-saroha) [inline app/controllers/api/v1/levels_controller.rb]: move to locale

File: config/routes.rb [hunk 4]
```diff
@@ -3,6 +3,8 @@
 Rails.application.routes.draw do
   namespace :api do
     namespace :v1 do
+      post "/city_sport_levels", to: "levels#all_levels"
```
Comment (tanya-saroha) [inline config/routes.rb]: Why is listing a 'post' request?

File: app/controllers/api/v1/levels_controller.rb (hunk 2 above)
Comment (DevKaranPradhan) [reply app/controllers/api/v1/levels_controller.rb]: moved error messages to locale under levels

File: .rubocop.yml (hunk 1 above)
Comment (DevKaranPradhan) [reply .rubocop.yml]: changed it back to 15

File: config/routes.rb (hunk 4 above)
Comment (DevKaranPradhan) [reply config/routes.rb]: changed it to get

**Pull request merged**